src/qibolab/instruments/qblox/cluster_qrm_rf.py

```diff
@@ -613,14 +613,18 @@ def process_pulse_sequence(
                     }
 
                 # Acquisitions
+                pulse = None
                 for acquisition_index, pulse in enumerate(sequencer.pulses.ro_pulses):
                     sequencer.acquisitions[pulse.serial] = {
                         "num_bins": num_bins,
                         "index": acquisition_index,
                     }
 
                 # Add scope_acquisition to default sequencer
-                if sequencer.number == self.DEFAULT_SEQUENCERS[port]:
+                if (
+                    sequencer.number == self.DEFAULT_SEQUENCERS[port]
+                    and not pulse == None
```

Comment:
Also, it is a single operation (instead of two), and less error-prone (it does not rely on the priority of operators).
But it's quite negligible, you could summarize saying that it is more idiomatic (aka Pythonic). The main feature is readability.